packages/x-components/src/x-modules/queries-preview/components/__tests__/query-preview-list.spec.ts

```diff
@@ -62,13 +59,9 @@ function renderQueryPreviewList({
     adapter,
     wrapper,
     queryPreviewListWrapper: wrapper.findComponent(QueryPreviewList),
-    getQueryPreviewItemWrappers() {
-      return wrapper.findAllComponents(QueryPreview);
-    },
-    reRender() {
-      // A timeout with no time should resolve after all reactivity + promises involved
-      return new Promise(resolve => setTimeout(resolve));
-    }
+    getQueryPreviewItemWrappers: () => wrapper.findAllComponents(QueryPreview),
+    // A timeout with no time should resolve after all reactivity + promises involved
+    reRender: () => new Promise(resolve => setTimeout(resolve))
```

Comment:
Would it not have the same effect as having `await nextTick()`?

Reply:
`+ promises involved` is the key. I replaced this function by `flushPromises` which in essence is the same we are doing in the `reRender`